lib/mongoid/validatable/uniqueness.rb

```diff
@@ -94,7 +94,9 @@ def case_sensitive?
       # @return [ Criteria ] The criteria.
       def create_criteria(base, document, attribute, value)
         criteria = scope(base.unscoped, document, attribute)
-        criteria.selector.update(criterion(document, attribute, value.mongoize))
+        field = document.fields[document.database_field_name(attribute)]
+        mongoized = field.try(:localized?) ? value.mongoize : field.mongoize(value) 
```

Comment:
Can you please add a comment here explaining the logic, specifically when is the instance mongoize the correct method to call and when the class one is the correct one?

Reply:
done